### ISSUE TITLE
The request for the training code

### ISSUE THREAD

**Issue by Lincoln20030413** [osmosis-diffusion/osmosis-diffusion-code#2]
Thanks for your great work "Osmosis". I am really interested in following your work and I have carefully read your shared source code on github. Regrettably, there are only the code of sampling and pretrained checkpoint of RGBD prior. However, the pretraining and finetuning part for RGBD prior in your work also attracts me. Could you share it?

**Comment by opherbambi** [comment]
Due to technical issues, I currently don't have access to the exact code.
The training code I use is from [here](https://github.com/openai/guided-diffusion) (OpenAI 'Guided-Diffusion' openai/guided-diffusion)

There is a great explanation on youtube - [link](https://www.youtube.com/watch?v=y7J6sSO1k50&list=PLBoQnSflObcnYM35qBaDFQjaJEh_5Wx-O&index=2) and [link](https://www.youtube.com/watch?v=hAp7Lk7W4QQ&list=PLBoQnSflObcnYM35qBaDFQjaJEh_5Wx-O&index=3). 
The only changes I did are:
> - Changing the first and the last layers to be 4 channels (instead of 3 channels)  
> - Training with RGBD datasets (instead of RGB)

**About the training**, we set T (number of steps) as 1000, as originally suggested in the original paper (OpenAI 'Guided-Diffusion' openai/guided-diffusion).
I don't remember the number of training iterations (unfortunately I don't have access to this information anymore), but it was trained for 7 days on 4 Nvidia A100 GPUs.

**About the data preprocessing**,  for **HR-SWI** and **ReDWeB** datasets, we normalized the values of each depth map to be from -1 (min value) to 1 (max value). The Reason is that those depth maps have no scale, because they were generated by algorithm or stereo cameras and were not scaled.
For **KITTI** and **DIODE** datasets, we set the maximum and minimum values to be -1 and 1 respectively (as described for those datasets), and scale the depth map values according to this max and min (I do not remember the values - check in their papers).

**About the depth masks** in the training process, we masked out pixels where there is no information (usually the far areas like the sky) from the loss function. It means that we don't calculate loss for those pixels.
But, in my opinion, the **depth masks** preprocess should include those "sky" areas.
Explain, usually the masks mask out the "sky" in outdoor scenes, therefore most of the "sky" areas in our reconstructed images are not reconstructed good enough (check Failure Cases in the appendix). 
The right thing to do, in my opinion, is following the handling of the masks of the depths like [here](https://depth-gen.github.io/). 
They preprocess the images by segmenting the "sky" areas and give them some maximum value, and did not mask them out from the loss function (as we did).

**Comment by c-pupil** [comment]
How should I set the hyperparameters to fine-tune **256x256_diffusion_uncond.pt** using my own dataset with the training code from [https://github.com/openai/guided-diffusion](url)?  Additionally, could you please share the training command? Thank you for your assistance! @opheroutsense 
Here are my own training commands, I don't know if it's correct.`python image_train.py --data_dir /home/xxx/dataset --attention_resolutions 32,16,8 --class_cond False --image_size 256 --learn_sigma True --num_channels 256 --num_res_blocks 2 --resblock_updown True --use_fp16 True --use_scale_shift_norm True --diffusion_steps 1000 --noise_schedule linear  --lr 1e-4 --batch_size 16 --save_interval 10000 --log_interval 1000  --resume_checkpoint /home/xxxl/guided-diffusion/models/256x256_diffusion_uncond.pt` Additionally, I found that I cannot train on multiple GPUs.
